src/ai/backend/common/clients/valkey_client/valkey_artifact_storages/client.py

```diff
@@ -0,0 +1,257 @@
+    async def set_object_storage(
+        self,
+        storage_id: uuid.UUID,
+        storage_data: ObjectStorageStatefulData,
+    ) -> None:
+        """
+        Cache object storage data.
+
+        :param storage_id: The UUID of the object storage.
+        :param storage_data: The storage data to cache.
+        """
+        key = self._make_storage_key(ArtifactStorageType.OBJECT_STORAGE, storage_id)
+        value = dump_json_str(dataclasses.asdict(storage_data))
+        await self._client.client.set(
+            key=key,
+            value=value,
+            expiry=ExpirySet(ExpiryType.SEC, _EXPIRATION),
+        )
+        log.debug("Cached object storage data for storage_id={}", storage_id)
+
+    @valkey_artifact_storages_resilience.apply()
+    async def get_object_storage(
+        self,
+        storage_id: uuid.UUID,
+    ) -> Optional[ObjectStorageStatefulData]:
+        """
+        Get cached object storage data.
+
+        :param storage_id: The UUID of the object storage.
+        :return: The cached storage data or None if not found.
+        """
+        key = self._make_storage_key(ArtifactStorageType.OBJECT_STORAGE, storage_id)
+        value = await self._client.client.get(key)
+        if value is None:
+            log.debug("Cache miss for object storage with storage_id={}", storage_id)
+            return None
+
+        json_value = value.decode()
+        data = load_json(json_value)
+        # Convert UUID string back to UUID object
+        if "id" in data and isinstance(data["id"], str):
+            data["id"] = uuid.UUID(data["id"])
+        return ObjectStorageStatefulData(**data)
+
+    @valkey_artifact_storages_resilience.apply()
+    async def delete_object_storage(
+        self,
+        storage_id: uuid.UUID,
+    ) -> bool:
+        """
+        Delete cached object storage data.
+
+        :param storage_id: The UUID of the object storage.
+        :return: True if the key was deleted, False otherwise.
+        """
+        key = self._make_storage_key(ArtifactStorageType.OBJECT_STORAGE, storage_id)
+        result = await self._client.client.delete([key])
+        deleted = result > 0
+        if deleted:
+            log.debug("Deleted cached object storage data for storage_id={}", storage_id)
+        return deleted
+
+    @valkey_artifact_storages_resilience.apply()
+    async def set_vfs_storage(
+        self,
+        storage_id: uuid.UUID,
+        storage_data: VFSStorageStatefulData,
+    ) -> None:
+        """
+        Cache VFS storage data.
+
+        :param storage_id: The UUID of the VFS storage.
+        :param storage_data: The storage data to cache.
+        """
+        key = self._make_storage_key(ArtifactStorageType.VFS_STORAGE, storage_id)
+        # Convert Path to string for JSON serialization
+        data_dict = dataclasses.asdict(storage_data)
+        data_dict["base_path"] = str(data_dict["base_path"])
+        value = dump_json_str(data_dict)
```

Comment:
If we're going to save it like this, let's just define it with pydantic instead.